lang/cs/Org.Apache.REEF.Common.Tests/metrics/DefaultMetricSourceTests.cs

```diff
@@ -0,0 +1,265 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Linq;
+using Org.Apache.REEF.Common.Metrics.Api;
+using Org.Apache.REEF.Common.Metrics.MutableMetricsLayer;
+using Org.Apache.REEF.Tang.Implementations.Tang;
+using Org.Apache.REEF.Tang.Interface;
+using Xunit;
+
+namespace Org.Apache.REEF.Common.Tests.Metrics
+{
+    public sealed class DefaultMetricSourceTests
+    {
+        /// <summary>
+        /// Tests <see cref="DefaultMetricsSourceConfiguration"/>
+        /// </summary>
+        [Fact]
+        public void TestMetricsSourceConfiguration()
+        {
+            const string evalId = "evalId";
+            const string taskId = "taskId";
+            const string sourceContext = "test";
+            const string recordName = "testrecord";
+
+            DefaultMetricsSourceImpl metricsSource = TangFactory.GetTang().NewInjector(
+                GenerateMetricsSourceConfiguration(evalId, taskId, sourceContext, recordName))
+                .GetInstance<DefaultMetricsSourceImpl>();
+
+            Assert.NotNull(metricsSource);
+
+            MetricTestUtils.MetricsCollectorTestImpl collector = new MetricTestUtils.MetricsCollectorTestImpl();
+            metricsSource.GetMetrics(collector, true);
+            Assert.Equal(collector.CurrentRecordBuilder.Name, recordName);
+            Assert.Equal(collector.CurrentRecordBuilder.Context, sourceContext);
+            collector.CurrentRecordBuilder.Validate("TaskOrContextName", taskId);
+            collector.CurrentRecordBuilder.Validate("EvaluatorId", evalId);
+        }
+
+        /// <summary>
+        /// Tests various functionalities of <see cref="DefaultMetricsSourceImpl"/>
```

Comment:
Done ... Also I have retained part of this test as a separate long test to make sure that adding different types of metrics do not lead to some weird bugs.